.github/workflows/extensions.yml

```diff
@@ -0,0 +1,170 @@
+  # workflow
+  publisher-command-center:
+    needs: [complex-extension-changes]
+    if: ${{ needs.complex-extension-changes.outputs.publisher-command-center == 'true' }}
+    uses: ./.github/workflows/publisher-command-center.yml
+
+  # Gathers all release data from GitHub releases triggered by this workflow
+  # For use in the `update-extension-list` job
+  # If no releases were triggered the output for releases will be `[]`
+  fetch-releases:
+    runs-on: ubuntu-latest
+    needs: [simple-extensions, publisher-command-center]
+    if: ${{ always() }}
+    outputs:
+      releases: ${{ steps.fetch-releases.outputs.releases }}
+
+    steps:
+      # Downloads every release data file from the release-extension action
+      # merging them under the .releases/ directory
+      - name: Download GitHub release data
+        uses: actions/download-artifact@v4
+        with:
+          pattern: release-*.json
+          path: releases
+          merge-multiple: true
+
+      # We use jq --slurp to create a single JSON array from all the JSON files
+      # to use in the `update-extension-list` job
+      - name: Fetch releases
+        id: fetch-releases
+        run: echo "releases=$(cat releases/*.json | jq -c --slurp .)" >> "$GITHUB_OUTPUT"
+
+  # Updates the `extensions.json` file with the latest release data from
+  # all extensions that were released in this workflow using the `fetch-releases`
+  # job output
+  update-extension-list:
+    runs-on: ubuntu-latest
+    needs: [fetch-releases]
+    # Only runs if there are releases to update the extension list with
+    # https://github.com/actions/runner/issues/2205
+    if: ${{ always() && needs.fetch-releases.result == 'success' && needs.fetch-releases.outputs.releases != '[]' }}
+    # Sets the RELEASES environment variable for the extension list update
+    # script to read in
+    env:
+      RELEASES: ${{ needs.fetch-releases.outputs.releases }}
+
+    steps:
+      # Checkout main to commit the updated extension list
+      # reduces the chance of conflicts when updating the extension list with
+      # multiple running workflows
+      - uses: actions/checkout@v4
+        with:
+          ref: main
+
+      - uses: actions/setup-node@v4
+        with:
+          node-version: "lts/*"
+          cache: "npm"
+          cache-dependency-path: scripts/package-lock.json
+
+      - run: npm ci
+        working-directory: ./scripts
+
+      - run: npm run update-extension-list
+        working-directory: ./scripts
+
+      # Commits and pushes the updated extension list to the repository
+      # https://github.com/actions/checkout/tree/v4/?tab=readme-ov-file#push-a-commit-using-the-built-in-token
+      - run: |
+          git config user.name "github-actions[bot]"
+          git config user.email "41898282+github-actions[bot]@users.noreply.github.com"
+          git add extensions.json
+          git commit -m "Update extension list"
+          git push
```

Comment:
Is this pushing directly to `main`? Should this go through a pull request to ensure a human is in the loop before changes make it to Connect?

Reply:
It is pushing directly to `main` yes. I could adjust it to instead open a PR.

My thought around this part of the flow is that a PR containing the updates to extensions has already been reviewed and merged and a release has already gone out in the repo, so the extension list should update right away.

I can see an argument though that any adjustments to `main` should go through a PR flow.

Reply:
That makes sense. I don't see anything wrong with moving forward with this approach. The risk seems minimal.